var/spack/repos/builtin/packages/py-chainer/package.py

```diff
@@ -34,3 +38,58 @@ class PyChainer(PythonPackage):
     depends_on('py-filelock', type=('build', 'run'))
     depends_on('py-protobuf@3:', type=('build', 'run'))
     depends_on('py-typing@:3.6.6', when='@:6', type=('build', 'run'))
+
+    # Dependencies only required for test of ChainerMN
+    depends_on('py-matplotlib', type=('build', 'run'), when='+mn')
+    depends_on('py-mpi4py', type=('build', 'run'), when='+mn')
+    depends_on("mpi", type=("build", "run"), when='+mn')
+
+    @run_after('install')
+    def cache_test_sources(self):
+        if '+mn' in self.spec:
+            self.cache_extra_test_sources("examples")
+
+    def test(self):
+        if "+mn" in self.spec:
+            # Run test of ChainerMN
+            test_dir = self.test_suite.current_test_data_dir
+
+            mnist_dir = join_path(
+                self.install_test_root, "examples", "chainermn", "mnist"
+            )
+            mnist_file = join_path(mnist_dir, "train_mnist.py")
+            mpi_name = self.spec["mpi"].prefix.bin.mpirun
+            python_exe = self.spec["python"].command.path
+            opts = [
+                "-n",
+                "4",
+                python_exe,
+                mnist_file,
+                "-o",
+                test_dir,
+            ]
+            env["OMP_NUM_THREADS"] = "4"
+
+            # set LD_PRELOAD for Fugaku
+            if (self.spec.target == 'a64fx' and
+                self.spec['mpi'].name == 'fujitsu-mpi'):
+                lib_path = join_path(
+                    "/usr", "lib", "FJSVtcs", "ple", "lib64", "libpmix.so"
+                )
+                if os.path.exists(lib_path):
+                    env["LD_PRELOAD"] = lib_path
+
+            self.run_test(
+                mpi_name,
+                options=opts,
+                work_dir=test_dir,
+            )
+
+            # check results
+            json_open = open(join_path(test_dir, 'log'), 'r')
+            json_load = json.load(json_open)
+            v = {d.get('epoch'): d.get('main/accuracy') for d in json_load}
```

Comment:
Dictionary comprehension are not allowed in Python 2.6 :cry: 
```suggestion
            v = dict([(d.get('epoch'), d.get('main/accuracy')) for d in json_load])
```

Reply:
Thank you for your kind suggestion.